corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/session_nav.js

```diff
@@ -18,80 +16,86 @@ FormplayerFrontend.module("SessionNavigate", function (SessionNavigate, Formplay
         selectApp: function (appId) {
             SessionNavigate.MenuList.Controller.selectMenu(appId);
         },
-        listMenus: function (appId) {
+        listMenus: function () {
             FormplayerFrontend.request("clearForm");
             var currentFragment = Backbone.history.getFragment();
-            var paramMap = Util.getSteps(currentFragment);
-            var steps = paramMap.steps;
-            var page = paramMap.page || 0;
-            var search = paramMap.search || "";
-            SessionNavigate.MenuList.Controller.selectMenu(appId, steps, page, search);
+            var urlObject = Util.CloudcareUrl.fromJson(Util.encodedUrlToObject(currentFragment));
+            var appId = urlObject.appId;
+            var sessionId = urlObject.sessionId;
+            var steps = urlObject.steps;
+            var page = urlObject.page;
+            var search = urlObject.search;
+            SessionNavigate.MenuList.Controller.selectMenu(appId, sessionId, steps, page, search);
         },
         showDetail: function (model, index) {
             SessionNavigate.MenuList.Controller.showDetail(model, index);
         },
-        listSessions: function() {
+        listSessions: function () {
             SessionNavigate.SessionList.Controller.listSessions();
         },
-
-        getIncompleteForm: function(sessionId) {
+        getIncompleteForm: function (sessionId) {
             FormplayerFrontend.request("getIncompleteForm", sessionId);
         },
+        renderResponse: function (menuResponse) {
```

Comment:
This is the part of the code that I like the least. Essentially we need a way to push the MenuResponse (returned as a parameter of the submission response) as if it was returned regularly by a request from the menu entity to the backend